test/cmd/command.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"knative.dev/pkg/test/helpers"
+)
+
+// RunCommand will run the command and return the standard output, plus error if there is one.
+func RunCommand(cmdLine string) (string, error) {
+	cmdSplit := strings.Fields(cmdLine)
+	if len(cmdSplit) == 0 {
+		return "", errors.New("the command line cannot be empty")
+	}
+
+	cmdName := cmdSplit[0]
+	args := cmdSplit[1:]
+	cmd := exec.Command(cmdName, args...)
+	var eb bytes.Buffer
+	cmd.Stderr = &eb
+
+	out, err := cmd.Output()
+	if err != nil {
+		errorCode := getErrorCode(err)
+		commandLineErr := CommandLineError{Command: cmdLine, ErrorOutput: eb.Bytes(), ErrorCode: errorCode}
+		return string(out), commandLineErr
+	}
+
+	return string(out), nil
+}
+
+// RunCommands will run the commands sequentially.
+// If there is an error when running a command, it will return directly with all standard output so far and the error.
+func RunCommands(cmdLines ...string) (string, error) {
+	var outputs []string
+	for _, cmdLine := range cmdLines {
+		output, err := RunCommand(cmdLine)
+		outputs = append(outputs, output)
+		if err != nil {
+			return strings.Join(outputs, "\n"), fmt.Errorf("error running %q: %v", cmdLine, err)
```

Comment:
I feel it redundant to change it that way, since the `outputs` array will be extended in the for loop. Or are you suggesting using a `StringBuilder` here?

Reply:
Indeed, it's cleaner this way, nevermind my comment.